exporter/prometheusremotewriteexporter/exporter_test.go

```diff
@@ -704,3 +705,146 @@ func Test_validateAndSanitizeExternalLabels(t *testing.T) {
 		})
 	}
 }
+
+// Ensures that when we attach the Write-Ahead-Log(WAL) to the exporter,
+// that it successfully writes the serialized prompb.WriteRequests to the WAL,
+// and that we can retrieve those exact requests back from the WAL, when the
+// exporter starts up once again, that it picks up where it left off.
+func TestWALOnExporterRoundTrip(t *testing.T) {
+	if testing.Short() {
+		t.Skip("This test could run for long")
+	}
+
+	// 1. Create a mock Prometheus Remote Write Exporter that'll just
+	// receive the bytes uploaded to it by our exporter.
+	uploadedBytesCh := make(chan []byte, 1)
+	exiting := make(chan bool)
+	prweServer := httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+		uploaded, err2 := ioutil.ReadAll(req.Body)
+		assert.Nil(t, err2, "Error while reading from HTTP upload")
+		select {
+		case uploadedBytesCh <- uploaded:
+		case <-exiting:
+			return
+		}
+	}))
+	defer prweServer.Close()
+
+	// 2. Create the WAL configuration, create the
+	// exporter and export some time series!
+	tempDir := t.TempDir()
+	cfg := &Config{
+		ExporterSettings: config.NewExporterSettings(config.NewComponentID(typeStr)),
+		Namespace:        "test_ns",
+		HTTPClientSettings: confighttp.HTTPClientSettings{
+			Endpoint: prweServer.URL,
+		},
+		RemoteWriteQueue: RemoteWriteQueue{NumConsumers: 1},
+		WAL: &WALConfig{
+			Directory:         tempDir,
+			TruncateFrequency: 60 * time.Microsecond,
```

Comment:
What are the implications of this? Wouldn't this be too hard on the CPU used in the CI machines? Perhaps 1ms wouldn't be too bad in this case?

Reply:
Or will it just truncate once?

Reply:
I've removed this as it doesn't seem to affect the tests.  I expect this is gated on the batch size instead of frequency.